### ISSUE TITLE
Add extension points to events

### ISSUE THREAD

**Issue by williamhogman** [open-broker/specification#39]
We're missing extensions for some events that need them.

**Comment by mantono** [comment]
@williamhogman What is the status on this? Should the ability to add extensions be added to more events or this the current order of things satisfactory?

**Comment by williamhogman** [comment]
I don't like the concept of Extensions too much to be honest. It kinda goes against the point of the spec. Perhaps having an extension point in the root of everything is all thats needed.

**Comment by mantono** [comment]
I agree that extension points should not be relied upon too much, but that's why I think they should *not* be in the root, because then, If I understand you right, they could occur in every event? I think this would be an disadvantage and would make it harder to conform to the spec. It would also be bit strange too always have (the possibility) of extensions in both the cloud event and the open broker event every time. I think they should either be in just an ApplicationCreated event or not exist at all.

**Comment by williamhogman** [comment]
Cloudevent extensions are completely unrelated those are to be used in the same manner as HTTP header. Open-broker extensions are meant to be used for fields that don't exist in the spec but are required by some service provider.
